src/Compilers/Core/CommandLine/ConsoleUtil.cs

```diff
@@ -41,7 +38,7 @@ internal static T RunWithUtf8Output<T>(Func<TextWriter, T> func)
 
         internal static T RunWithUtf8Output<T>(bool utf8Output, TextWriter textWriter, Func<TextWriter, T> func)
         {
-            if (utf8Output)
+            if (utf8Output && textWriter.Encoding.CodePage != s_utf8Encoding.CodePage)
```

Comment:
This is important for running tests - we manually invoke with a utf8 `TextWriter` when we pass `/utf8output`, so this doesn't hit the true case (because it then would fail, as the textwriter is not `Console.Out`)